dtfs-central-api/src/services/state-machines/utilisation-report/event-handlers/remove-fees-from-payment/remove-fees-from-payment.event-handler.test.ts

```diff
@@ -0,0 +1,115 @@
+import { EntityManager } from 'typeorm';
+import { UtilisationReportEntityMockBuilder, DbRequestSource, FeeRecordEntityMockBuilder, UtilisationReportEntity } from '@ukef/dtfs2-common';
+import { handleUtilisationReportRemoveFeesFromPaymentEvent } from './remove-fees-from-payment.event-handler';
+import { FeeRecordStateMachine } from '../../../fee-record/fee-record.state-machine';
+import { feeRecordsMatchAttachedPayments } from '../helpers';
+
+jest.mock('../helpers');
+
+describe('handleUtilisationReportRemoveFeesFromPaymentEvent', () => {
+  const tfmUserId = 'abc123';
+  const requestSource: DbRequestSource = {
+    platform: 'TFM',
+    userId: tfmUserId,
+  };
+
+  const mockSave = jest.fn();
+  const mockEntityManager = {
+    save: mockSave,
+  } as unknown as EntityManager;
+
+  const aReconciliationInProgressReport = () => UtilisationReportEntityMockBuilder.forStatus('RECONCILIATION_IN_PROGRESS').build();
+
+  const aFeeRecordForReport = (report: UtilisationReportEntity, id: number) => FeeRecordEntityMockBuilder.forReport(report).withId(id).build();
+
+  const aMockEventHandler = () => jest.fn();
+  const aMockFeeRecordStateMachine = (eventHandler: jest.Mock): FeeRecordStateMachine =>
+    ({
+      handleEvent: eventHandler,
+    }) as unknown as FeeRecordStateMachine;
+
+  beforeEach(() => {
+    jest.spyOn(FeeRecordStateMachine, 'forFeeRecord').mockReturnValue(aMockFeeRecordStateMachine(aMockEventHandler()));
+  });
+
+  afterEach(() => {
+    jest.resetAllMocks();
+  });
+
+  it.each([
+    { feeRecordsAndPaymentsMatch: true, testNameSuffix: 'the fee records match the payments' },
+    { feeRecordsAndPaymentsMatch: false, testNameSuffix: 'the fee records do not match the payments' },
+  ])(
+    "calls the fee record state machine event handlers with 'feeRecordsAndPaymentsMatch' set to '$feeRecordsAndPaymentsMatch' if $testNameSuffix",
```

Comment:
I'm not sure the test name matches the assertion, reading this name I would expect us to just be asserting the specific field, but for the ones we are removing we don't even have this parameter? Perhaps we are attempting to test too much in one test?

Reply:
Split this into two test cases now to test that the handlers for the two fee record events `REMOVE_FROM_PAYMENT_GROUP` and `OTHER_FEE_REMOVED_FROM_GROUP` are called - changes in a94cf7f80e3f741b30aa42bcd2c597ecd95c99ef thanks